cupy/linalg/_decomposition.py

```diff
@@ -352,6 +354,63 @@ def qr(a, mode='reduced'):
     return q, _util._triu(r)
 
 
+def _svd_batched(a, a_dtype, full_matrices, compute_uv):
+    batch_shape = a.shape[:-2]
+    batch_size = internal.prod(batch_shape)
+    n, m = a.shape[-2:]
+    s_dtype = a_dtype.lower()
+
+    # first handle any 0-size inputs
+    if batch_size == 0:
+        k = min(m, n)
+        s = cupy.empty(batch_shape + (k,), s_dtype)
+        if compute_uv:
+            if full_matrices:
+                u = cupy.empty(batch_shape + (n, n), dtype=a_dtype)
+                vt = cupy.empty(batch_shape + (m, m), dtype=a_dtype)
+            else:
+                u = cupy.empty(batch_shape + (n, k), dtype=a_dtype)
+                vt = cupy.empty(batch_shape + (k, m), dtype=a_dtype)
+            return u, s, vt
+        else:
+            return s
+    elif m == 0 or n == 0:
+        s = cupy.empty(batch_shape + (0,), s_dtype)
+        if compute_uv:
+            if full_matrices:
+                u = [cupy.eye(n, dtype=a_dtype) for _ in range(batch_size)]
+                u = cupy.stack(u, axis=0).reshape(*batch_shape, n, n)
```

Comment:
How about
```suggestion
                u = cupy.ascontiguousarray(cupy.broadcast_to(
                    cupy.identity(n, a_dtype),
                    batch_shape + (n, n)))
```
or
```suggestion
                u = cupy.empty(batch_shape + (n, n), a_dtype)
                u[...] = cupy.identity(n, a_dtype)
```

My thought on other possible implementations:

- `cupy.stack([cupy.eye(n, dtype=a_dtype)] * batch_size, axis=0).reshape(*batch_shape, n, n)` reduces the number of calls of `eye`, but still creates a list.
- `cupy.eye(n, dtype=a_dtype)[None].repeat(batch_size, axis=0).reshape(*batch_shape, n, n)` might not be fast because `repeat` supports
  ```
  >>> cupy.repeat(cupy.arange(3), [3, 2, 1])
  array([0, 0, 0, 1, 1, 2])
  ```
- `cupy.tile(cupy.eye(n, dtype=a_dtype)[None], (batch_size, 1, 1)).reshape(*batch_shape, n, n)` is somewhat verbose.

Reply:
I think your second suggestion of using `cupy.identity` seems nice, I'll test it locally. Note that this is a very special (that is, silly) call path that in reality would be unlikely to hit, and I don't think performance matters here...

Reply:
Updated in 5a0376a.